### PR TITLE
Feature/user

### PR DESCRIPTION
**Pull request by hugh-n**

### CODE REVIEW

File: app/assets/stylesheets/object/user/home/article-list.scss [hunk 1]
```diff
@@ -1,38 +1,218 @@
...
+
+//urls
+$url_0: url(https://fonts.googleapis.com/css?family=Open+Sans:600);
+$url_1: url(http://xendox.com/assets/img/header_bg.jpg);
+
+//other vars
+$radius: 100px;
+
+%extend_1 {
+	height: 100%;
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/home/article-list.scss]: Please write from A to Z all in this file.

File: app/assets/stylesheets/layout/user/_header.scss [hunk 2]
```diff
@@ -1,20 +1,116 @@
...
+      width: 100%;
+    }
+  }
+}
+
+.p-users-navbar{
+  background-color: #62BA69;
+  overflow: hidden;
+  a{
+    float: left;
```
Comment (ericgpks) [inline app/assets/stylesheets/layout/user/_header.scss]: Please write from A to Z.

File: app/views/users/registrations/new.html.haml [hunk 3]
```diff
@@ -1,74 +1,51 @@
...
             %label{for: "user_family_name"} 姓(全角漢字)
-            = text_field :user, :family_name, class: "form-control", placeholder: "山田"
+            = text_field :user, :family_name, class: "form-control form-name", placeholder: "山田", "data-name": "名前(姓)"
+            #err-message-user_family_name
           .form-group.col-md-6
             %label{for: "user_given_name"} 名(全角漢字)
-            = text_field :user, :given_name, class: "form-control", placeholder: "太郎"
-
-        .form-row
+            = text_field :user, :given_name, class: "form-control form-name", placeholder: "太郎", "data-name": "名前(名)"
```
Comment (ericgpks) [inline app/views/users/registrations/new.html.haml]: When you use "data-" in haml, you can write like this.
https://qiita.com/k6i/items/e2f2675dd6e3c2212c1b

File: app/assets/stylesheets/service.scss [hunk 4]
```diff
@@ -24,28 +26,153 @@
...
+  }
+  &__orange:hover, .btn-orange:focus {
     color: #fff;
     box-shadow: 5px 5px 4px 0px rgba(0, 0, 0, 0.4)
 	
-}
-.service_title{
+  }
+  &__btn-service {
+    border: 1px solid #62BA69;
```
Comment (ericgpks) [inline app/assets/stylesheets/service.scss]: Please write each element in order from A to Z all in this file.

File: app/controllers/service_controller.rb [hunk 5]
```diff
@@ -7,7 +7,7 @@ def index
 			@name = @check.name ## adding name for display
 			@business =  Business.joins("LEFT JOIN business_services ON business_services.business_id = businesses.id LEFT JOIN m_services ON business_services.service_id = m_services.id  LEFT JOIN m_service_categories ON m_services.category_id = m_service_categories.id ").select("businesses.*").where("m_service_categories.name_alias = '#{params["category_id"]}'").group("businesses.id") ## getting data by category
 			@count =  Business.joins("LEFT JOIN business_services ON business_services.business_id = businesses.id LEFT JOIN m_services ON business_services.service_id = m_services.id  LEFT JOIN m_service_categories ON m_services.category_id = m_service_categories.id ").where("m_service_categories.name_alias = '#{params["category_id"]}'").group("m_service_categories.id").count("businesses.id")  ## counting total data
-			@service = MService.where(:category_id => @check.id)
+			@services = MService.where(:category_id => @check.id)
```
Comment (ericgpks) [inline app/controllers/service_controller.rb]: Please do not use this style.
" =>"
You can change this to "category_id: @check.id".

File: app/views/elements/service/_businessdata.html.haml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.p-i-service__body1
+  .p-i-service__container.container
+    %ul.p-i-service__unstyled.centered
+      %li.p-i-service__li
+        = check_box_tag "select_all" ,'1', false,  class: "p-i-service__styled-checkbox"
+        %label.checkbox{ :for => "select_all" , :class => "p-i-service__checkbox_label"}
```
Comment (ericgpks) [inline app/views/elements/service/_businessdata.html.haml]: Please do not use this style.
"=>"
You can change this to { for: "select_all", class: "p-i-service__checkbox_label" }

File: app/views/elements/service/_countservice.html.haml [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+      .p-i-service__des
+        サービス選定にお迷いの場合は、
+        %strong.p-i-service__des-bold
+          まとめて資料請求
+        で各社を一度に比較しましょう！<br>
+        %strong.p-i-service__des-bold
+          = "#{@name}のサービス資料（1点）"
+        をまとめてダウンロードできます。<br>ダウンロードの際にご登録された個人情報は資料ダウンロード元のサービス運営会社（その連携会社）に提供されます。
+        .p-i-service__link
+          = link_to "まとめて資料請求（12件）", "/admin/businesses/", class: "btn p-i-service__orange"
```
Comment (ericgpks) [inline app/views/elements/service/_countservice.html.haml]: This figure might be variational.

File: app/views/elements/service/_countservice.html.haml [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
+    %h1.p-i-service__title
+      = @name+"の一覧"
+    .p-i-service__count_wrapper
+      .p-i-service__des
+        サービス選定にお迷いの場合は、
+        %strong.p-i-service__des-bold
+          まとめて資料請求
+        で各社を一度に比較しましょう！<br>
+        %strong.p-i-service__des-bold
+          = "#{@name}のサービス資料（1点）"
```
Comment (ericgpks) [inline app/views/elements/service/_countservice.html.haml]: This figure might be variational.

File: app/controllers/users/companies_controller.rb [hunk 9]
```diff
@@ -1,10 +1,10 @@
 class Users::CompaniesController < ApplicationController
-  before_action :authenticate_user!
 
   def index
   end
 
   def show
-    @company = Company.find(:params[:id])
+    @company = Company.find(params[:id])
+    @services = CompanyServices.joins(:m_service).where(company_id: params[:id])
```
Comment (ericgpks) [inline app/controllers/users/companies_controller.rb]: Is the class of CompanyService multiple?

File: app/assets/stylesheets/object/user/home/info-list.scss [hunk 10]
```diff
@@ -2,56 +2,108 @@
...
-    text-transform: uppercase;
-  }
-  &__article{
-    color: #62BA69;
-    font-family: "Roboto", sans-serif;
-    font-size: 15px;
-    float: center;
-    text-decoration: justify;
+.p-i-container-1 {
+  height: auto;
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/home/info-list.scss]: Please write each element from A to Z all in this file.

File: app/assets/stylesheets/layout/user/_header.scss [hunk 11]
```diff
@@ -14,10 +14,32 @@
...
+    &__nav{
+      margin-top: 20px;
+    }
     &__text{
       display:none;
     }
   }
+  .p-users-navbar{
+    a{
+      float: left;
```
Comment (ericgpks) [inline app/assets/stylesheets/layout/user/_header.scss]: Please write each element from A to Z all in this file.

File: app/views/elements/service/_businessdata.html.haml [hunk 12]
```diff
@@ -1,7 +1,27 @@
...
\ No newline at end of file
+          %span.p-i-service__checkbox_span 全てにチェックする（12件のサービスがあります）
+    %table.table.table-striped.p-i-service__table
+      - @businesses.each do |business|
+        %tr
+          %td.table-striped.p-i-service__table-checkbox
+            %ul.p-i-service__unstyled.centered
+              %li.p-i-service__li
+                = check_box_tag "select_one" ,'1', false,  class: "p-i-service__styled-checkbox"
+                %label.checkbox{ :for => "select_one" , :class => "p-i-service__checkbox_label"} &nbsp
```
Comment (ericgpks) [inline app/views/elements/service/_businessdata.html.haml]: Please do not use this style.
"=>"
You can change this to { for: "select_one", class: "p-i-service__checkbox_label" }

File: app/views/elements/user/companies/_countservice.html.haml [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    .p-i-service__count_wrapper
+      .p-i-service__des
+        この企業のサービスを
+        %strong.p-i-service__des-bold
+          まとめて資料請求
+        しましょう！<br>
+        %strong.p-i-service__des-bold
+          ダウンロードの際にご登録された個人情報は資料ダウンロード元のサービス運営会社（その連携会社）に提供されます。
+        .p-i-service__link
+          = link_to "まとめて資料請求（4件）", "/", class: "btn p-i-service__orange"
```
Comment (ericgpks) [inline app/views/elements/user/companies/_countservice.html.haml]: This figure might be variational.

File: app/assets/stylesheets/object/user/home/info-list.scss [hunk 14]
```diff
@@ -0,0 +1,106 @@
+@import "neat";
+
+$p-i-custom-grid--thirds: (
+  columns: 3,
```
Comment (ericgpks) [inline app/assets/stylesheets/object/user/home/info-list.scss]: Please write each element from A to Z all in this file.

File: app/views/users/home/index.html.haml [hunk 15]
```diff
@@ -1,38 +1,119 @@
...
+      - @businesses.each do |business|
+        %a{href: "", class: "p-i-home-business-link"}
+          .p-i-home-business
+            .p-i-home-business__logo
+              - if business.logo.nil?
+                %img{src:""}
+              - else
+                %img{src:"#{business.logo.url}"}
+            .p-i-home-business__title
+              #{business.title}
```
Comment (ericgpks) [inline app/views/users/home/index.html.haml]: When you show data you get from the controller, you can write like this.
"= business.title"

File: app/views/users/home/index.html.haml [hunk 16]
```diff
@@ -1,38 +1,119 @@
...
+            .p-i-home-business__explanation
+              #{business.explanation}
```
Comment (ericgpks) [inline app/views/users/home/index.html.haml]: "= business.explanation"

File: app/views/users/home/index.html.haml [hunk 17]
```diff
@@ -1,38 +1,119 @@
...
+            .p-i-home-business__services
+              - if !business.m_services.nil?
+                - business.m_services.each do |service|
+                  %span
+                    %i.fa.fa-tag
+                    #{service.name}
```
Comment (ericgpks) [inline app/views/users/home/index.html.haml]: "= service.name"

File: app/views/users/home/index.html.haml [hunk 18]
```diff
@@ -1,38 +1,119 @@
...
+    - if !@articles.nil?
+      - @articles.each do |article|
+        .p-i-home-articles__contents
+          .p-i-home-articles__logo
+            - if article.eye_catching_image.nil?
+              %img{src:""}
+            - else
+              %img{src:"#{article.eye_catching_image.url}"}
+          .p-i-home-business__title
+            #{article.title}
```
Comment (ericgpks) [inline app/views/users/home/index.html.haml]: "= article.title"

File: app/views/users/home/index.html.haml [hunk 19]
```diff
@@ -1,38 +1,119 @@
...
+          .p-i-home-business__explanation
+            #{article.content}
```
Comment (ericgpks) [inline app/views/users/home/index.html.haml]: "= article.content"

File: app/views/users/home/index.html.haml [hunk 20]
```diff
@@ -1,38 +1,119 @@
...
+          .p-i-home-business__services
+            - if !article.m_services.nil?
+              - article.m_services.each do |service|
+                %span
+                  %i.fa.fa-tag
+                  #{service.name}
```
Comment (ericgpks) [inline app/views/users/home/index.html.haml]: "= service.name"

File: app/views/users/home/index.html.haml (hunk 16 above)
Comment (hugh-n) [reply app/views/users/home/index.html.haml]: Fixed in local

File: app/views/users/home/index.html.haml (hunk 17 above)
Comment (hugh-n) [reply app/views/users/home/index.html.haml]: Fixed in local

File: app/views/users/home/index.html.haml (hunk 18 above)
Comment (hugh-n) [reply app/views/users/home/index.html.haml]: Fixed in local

File: app/views/users/home/index.html.haml (hunk 19 above)
Comment (hugh-n) [reply app/views/users/home/index.html.haml]: Fixed in local

File: app/views/users/home/index.html.haml (hunk 20 above)
Comment (hugh-n) [reply app/views/users/home/index.html.haml]: Fixed in local

File: app/models/user.rb [hunk 21]
```diff
@@ -1,6 +1,9 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  has_one :m_industry
+  has_one :m_company_size
+  has_one :m_department
+  has_one :m_roles
```
[the remark this answers is not in the document]
Comment (hugh-n) [reply app/models/user.rb]: Fixed in local.

File: app/models/users_token.rb [hunk 22]
```diff
@@ -0,0 +1,4 @@
+class UsersToken < ApplicationRecord
+  belongs_to :users
```
[the remark this answers is not in the document]
Comment (hugh-n) [reply app/models/users_token.rb]: Fixed in local.

File: app/controllers/users/companies_controller.rb (hunk 9 above)
Comment (hugh-n) [reply app/controllers/users/companies_controller.rb]: Changed to CompanyService. Fixed in local.

File: app/controllers/service_controller.rb [hunk 23]
```diff
@@ -0,0 +1,25 @@
+class ServiceController < ApplicationController
+	def index
+		@check = MServiceCategory.find_by(:name_alias => params["category_id"])
```
Comment (ericgpks) [inline app/controllers/service_controller.rb]: Please do not use this style.
"=>"
You can change this to "(name_alias: params["category_id"])

**Pull request closed without merging**